server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -371,7 +374,12 @@ protected QueryDataSet processDataQuery(QueryPlan queryPlan, QueryContext contex
       } else if (queryPlan instanceof LastQueryPlan) {
         queryDataSet = queryRouter.lastQuery((LastQueryPlan) queryPlan, context);
       } else {
-        queryDataSet = queryRouter.rawDataQuery((RawDataQueryPlan) queryPlan, context);
+        try {
+          queryDataSet = queryRouter.rawDataQuery((RawDataQueryPlan) queryPlan, context);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Sorry, I'm a little confused about when will this method throw a IndexOutOfBoundsException?

Reply:
When we have some nonsense predicates, such as 
 - select status, temperature from root.ln.wf01.wt01 where time < 2017-11-01T00:05:00.000 and time > 2017-11-01T00:12:00.000

Reply:
I believe this is a misuse, since it does not involve any index, and I would prefer creating a new Exception like `IllegalTimeRangeException` of `IllegalPredicateException`. You may correct it within this PR or just create an issue about it.

Reply:
To make it more clear, I created EmptyIntervalException.